packages/flutter/test/cupertino/segmented_control_test.dart

```diff
@@ -0,0 +1,651 @@
+        CupertinoColors.activeBlue);
+  });
+
+  testWidgets('Segmented control semantics', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = const Text('Child 1');
+    children[1] = const Text('Child 2');
+    int sharedValue = 0;
+
+    await tester.pumpWidget(
+      new StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return Directionality(
+            textDirection: TextDirection.ltr,
+            child: new Center(
+              child: SegmentedControl<int>(
+                children: children,
+                onValueChanged: (int newValue) {
+                  setState(() {
+                    sharedValue = newValue;
+                  });
+                },
+                groupValue: sharedValue,
+              ),
+            ),
+          );
+        },
+      ),
+    );
+
+    expect(
+        semantics,
+        hasSemantics(
+          new TestSemantics.root(
+            children: <TestSemantics>[
+              new TestSemantics.rootChild(
+                label: 'Child 1',
+                flags: <SemanticsFlag>[
+                  SemanticsFlag.isInMutuallyExclusiveGroup,
+                  SemanticsFlag.isSelected,
+                ],
+                actions: <SemanticsAction>[
+                  SemanticsAction.tap,
+                ],
+              ),
+              new TestSemantics.rootChild(
+                label: 'Child 2',
+                flags: <SemanticsFlag>[
+                  SemanticsFlag.isInMutuallyExclusiveGroup,
+                ],
+                actions: <SemanticsAction>[
+                  SemanticsAction.tap,
+                ],
+              ),
+            ],
+          ),
+          ignoreId: true,
+          ignoreRect: true,
+          ignoreTransform: true,
+        ));
+
+    await tester.tap(find.text('Child 2'));
+    await tester.pump();
+
+    expect(
+        semantics,
+        hasSemantics(
+          new TestSemantics.root(
+            children: <TestSemantics>[
+              new TestSemantics.rootChild(
+                label: 'Child 1',
+                flags: <SemanticsFlag>[
+                  SemanticsFlag.isInMutuallyExclusiveGroup,
+                ],
+                actions: <SemanticsAction>[
+                  SemanticsAction.tap,
+                ],
+              ),
+              new TestSemantics.rootChild(
+                label: 'Child 2',
+                flags: <SemanticsFlag>[
+                  SemanticsFlag.isInMutuallyExclusiveGroup,
+                  SemanticsFlag.isSelected,
+                ],
+                actions: <SemanticsAction>[
+                  SemanticsAction.tap,
+                ],
+              ),
+            ],
+          ),
+          ignoreId: true,
+          ignoreRect: true,
+          ignoreTransform: true,
+        ));
+
+    semantics.dispose();
+  });
+}
```

Comment:
golden file tests would be great here. you could test that it looks right with weird children, with pressed and selected states, etc.